app/Http/Controllers/HomeController.php

```diff
@@ -42,7 +45,46 @@ public function bbcodePreview()
 
     public function getChangelog()
     {
-        return view('home.changelog');
+        $build = presence(Request::input('build'));
+
+        $changelogs = Changelog::default();
+
+        if ($build !== null) {
+            $build = Build::where('version', $build)->firstOrFail();
+
+            $changelogs->where('build', $build->version);
+        } else {
+            $from = Changelog::default()->first();
+            $changelogs->where('date', '>', Carbon::parse($from->date)->subWeeks(config('osu.changelog.recent_weeks')));
+        }
+
+        $changelogs = $changelogs
+            ->with(['gameBuild', 'user'])
+            ->orderBy('major', 'desc')
+            ->get();
+
+        if ($build === null) {
```

Comment:
uh, but this is operating on the already retrieved collection, while during the previous conditional the query hasn't finished yet